### DISCUSSION TITLE
dozer cloud cli

### DISCUSSION THREAD

**Discussion by mediuminvader** [getdozer/dozer#1469] [Feature Requests]
The purpose of this discussion is to define the `dozer cloud` cli behavior.
The content will be updated according to the discussion.

the `admin.proto` definition is the reference: https://github.com/getdozer/dozer/blob/main/dozer-types/protos/admin.proto

# Core

## init
```sh
dozer init 
```
Bootstrap an Application project, creating the required files

# Cloud

## login
```sh
dozer cloud login -h hostname
```
Open a browser where the user can authenticate and returns a token to be used in the cli interaction.
The `hostname` is stored and reused along the session.

## deploy
```sh
dozer cloud deploy 
```
Create and Start an application remotely (executing a migration if needed).

## list
```sh
dozer cloud list
```
return the list of the Applications.

## delete
```sh
dozer cloud delete [--app id ]
```
Delete the Application identified by `id`.

## monitor
```sh
dozer cloud monitor [--app id ]
```
Print the progress of the Pipeline.

## logs
```sh
dozer cloud logs [--app id]  [-t]
```
Return the logs of the Application.

## connections
List connections
```sh
dozer cloud connections 
```
Show connection
```sh
dozer cloud connections show [connection_id] 
```

## validate
```sh
dozer cloud validate --app [app_name|app_id] 
```
Runs all validation on the specified app.

## status
```sh
dozer cloud status --app [app_name|app_id]
```
Return the status of an Application

## stop
```sh
dozer cloud stop --app [app_name|app_id]
```
Stop an Application

## Cloud vs Local
Almost all cloud commands should work with local instance as well.

**Comment by chubei** [comment]
I think we also need

`dozer cloud delete` to remove an application fully.
